instrumentation/opentelemetry-instrumentation-falcon/src/opentelemetry/instrumentation/falcon/__init__.py

```diff
@@ -169,6 +178,15 @@ def __init__(self, *args, **kwargs):
         self._excluded_urls = get_excluded_urls("FALCON")
         super().__init__(*args, **kwargs)
 
+    def _handle_exception(
+        self, req, resp, ex, params
+    ):  # pylint: disable=C0103
+        # Falcon 3 does not execute middleware within the context of the exception
+        # so we capture the exception here and save it into the env dict
+        _, exc, _ = exc_info()
+        req.env[_ENVIRON_EXC] = exc
+        return super()._handle_exception(req, resp, ex, params)
```

Comment:
I couldn't figure out another way to do this other than accessing this private method, FYI